core/controllers/base.py

```diff
@@ -253,9 +253,9 @@ def dispatch(self):
                 b'https://oppiatestserver.appspot.com', permanent=True)
             return
 
-        if (feconf.ENABLE_MAINTENANCE_MODE
-                and not self.current_user_is_super_admin
-                and self.request.path not in AUTH_HANDLER_PATHS):
+        if feconf.ENABLE_MAINTENANCE_MODE and (
+                self.request.path not in AUTH_HANDLER_PATHS
+                and not self.current_user_is_site_maintainer):
```

Comment:
I also think that it might make sense to move this whole thing into a function, smth like `is_maintenance_applied`?

Reply:
`is_maintenance_applied` doesn't explain what's going inside this function, I think the current structure is readable, I've also checked with Brian on whether we should have a [separate function for this or not ](https://github.com/oppia/oppia/pull/12884#discussion_r640015402)

Reply:
Hmm it might still make sense to add another wrapper here, something like `is_site_accessible_to_current_user()`?

Reply:
`is_site_accessible_to_current_user` souldss great, I'll add it as private function*

Reply:
As per offline discussion with @brianrodr, I've changed it to `_is_requested_path_currently_accessible_to_user`. Done!

Reply:
Thanks!